spharpy/samplings/coordinates.py

```diff
@@ -310,3 +361,35 @@ def n_max(self, value):
             self._n_max = None
         else:
             self._n_max = int(value)
+
+    @property
+    def radius_atol(self):
+        """Get or set the radius tolerance in meter."""
+        return self._radius_atol
+
+    @radius_atol.setter
+    def radius_atol(self, value):
+        """Get or set the radius tolerance in meter."""
+        if isinstance(value, (int, float)) and value > 0:
+            self._radius_atol = float(value)
+        else:
+            raise ValueError(
+                'The radius tolerance must be a number greater than zero')
+
+    def _set_points(self, x, y, z):
+
+        # check for unique radius
```

Comment:
instead of copy and pasting the code from pyfar we could just use the _check_points method here before checking the radius. this will do all common checks without setting the results. this will prevent the bug indicated by @mberz and we wouldnt have duplicated code

https://github.com/pyfar/pyfar/pull/803

Reply:
done